Vue3 migration

manuelmeister opened the pull request:
See #8271 for todos

Deployment: https://pr8279.ecamp3.ch/

BacLuc committed:
frontend: remove @ from login.infoText.dev in en,fr,it,rm

BacLuc committed:
Revert "frontend: disable validation for now in campCreate dialog"

This reverts commit bf4be475

BacLuc committed:
frontend: fix i18n in ServerErrorContent.vue

BacLuc committed:
frontend: fix i18n in toasts.js

BacLuc committed:
frontend: fix i18n in ButtonNestedContentNodeAdd

BacLuc committed:
frontend: fix i18n in CampListItem

BacLuc committed:
frontend: fix i18n in generatePdfMixin.js

BacLuc committed:
frontend: fix i18n in PrintPreviewClient.vue

BacLuc committed:
frontend: fix browserPreferredLocale in Login and Register

I did not find any docs to $i18n.browserPreferredLocale,
and no docs that it was removed.
But it's anyway better to use the browser standard.

BacLuc committed:
frontend: fix language switching in profile

Does not work yet on the register page. (But maybe never worked there).
Found this with the .value here: https://vue-i18n.intlify.dev/guide/essentials/scope#local-scope-1

BacLuc committed:
frontend: fix validation in DialogForm

Don't know if this correct or good, but it works.

BacLuc committed:
frontend: fix reference errors in own veeValidate rules

They don't work yet, but there is no referenceError.

BacLuc committed:
frontend: fix locale.replaceAll is not a function

BacLuc committed:
fixup! frontend: fix reference errors in own veeValidate rules

The second parameter was an array after all.

BacLuc committed:
frontend: readd validation to camp create dialog

BacLuc committed:
frontend: fix time validation in FormScheduleEntryItem

The value given in is now a datetime string and not anymore a time.

BacLuc committed:
frontend: fix selecting an item in ETimeDropdown

BacLuc committed:
frontend: apply changes for DialogForm to DialogBottomSheet

It doesn't work yet, but it should look similar in the end.

BacLuc committed:
frontend: change :value to :model-value in DetailPane

BacLuc committed:
frontend: fix slot passing in DialogBottomSheet variant for DetailPane

BacLuc committed:
frontend: fix referenceError in oneEmojiOrTwoCharacters.js

BacLuc committed:
frontend: strip reactive parts from serialized print config in print component

This is just a quick hack.
The correct way would be to not have reactive parts in the print config.

BacLuc committed:
frontend: fix vue-toastification in frontend print part

BacLuc committed:
frontend: fix PagesOverview

This was done with AI, but it seems to work somehow.
We need to check if we really need all changes.

BacLuc committed:
frontend: fix TypeError: can't access property "split", path is null in Login.vue

Until we figured out something better than setting path everywhere.

BacLuc committed:
frontend: fix falling back to global scope warning in Login.vue

BacLuc committed:
frontend: replace v-list-item-avatar with v-avatar in DialogActivityCreate

See https://github.com/vuetifyjs/vuetify/discussions/15553

BacLuc committed:
frontend: use tailwind like classes to center button in ClipboardInfoDialog

Vue warned about the unknown element center.

BacLuc committed:
frontend: enable vue/no-deprecated-dollar-listeners-api eslint rule

BacLuc committed:
frontend: set vue/no-deprecated-delete-set eslint to error

BacLuc committed:
frontend: enable vue/no-deprecated-destroyed-lifecycle eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-dollar-scopedslots-api eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-filter eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-props-default-this eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-slot-attribute eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-slot-scope-attribute eslint rule

BacLuc committed:
frontend: enable vue/no-deprecated-v-on-native-modifier eslint rule

BacLuc committed:
frontend: enable vue/no-v-for-template-key-on-child eslint rule

BacLuc committed:
frontend: enable 'vue/no-v-model-argument' eslint rule

BacLuc committed:
frontend: wrap ERichtext with Field

Like the other inputs.

BacLuc committed:
frontend: improve materialTable forms

You can now create material items in the material list overview.

BacLuc committed:
frontend: remove v-stepper-content in CampCreateStep

The element does not exist anymore with vuetify 3

BacLuc committed:
frontend: fix greaterThan.spec.js

manuelmeister committed:
style: fix navbar

manuelmeister committed:
style: fix container spacing

manuelmeister committed:
style: fix picasso scroller

manuelmeister committed:
nav: fix redirect

manuelmeister committed:
style: fix add section button

manuelmeister committed:
style: fix toolbar title

manuelmeister committed:
head: add old suffix

manuelmeister committed:
picasso: remove weekday

pmattmann committed:
fix user-menu top right

manuelmeister committed:
draggable: fix item-key

manuelmeister committed:
draggable: fix item-key

BacLuc committed:
Merge remote-tracking branch 'upstream/devel' into vue3

# Conflicts:
#	docker-compose.yml

BacLuc committed:
frontend: revert changes in frontend from devel

pmattmann committed:
PersonalInvitations, fix layout

simfeld committed:
Fix hyphens typo

pmattmann committed:
fix t$ calls

simfeld committed:
Make faq button visible

BacLuc committed:
frontend: allow to open the collaborator form

It does not feel right, but it works.

BacLuc committed:
frontend: allow to open the collaborator form for existing collaborations

It does not feel right, but it works.

BacLuc committed:
frontend: allow to deactivate collaborators

BacLuc committed:
frontend: fix PromptEntityDelete

BacLuc committed:
frontend: comment out SideBarProgram

It doesn't work right now.
When everything else works we can spend time fixing this.

BacLuc committed:
frontend: fix selects for collaboratorForm

Use our own map because v-select allows passing an object to the item slot,
but not to the selection slot.
So we use our own map as we did in the ActivityForm.

carlobeltrame committed:
Restore outlined style for api inputs

carlobeltrame committed:
Fix theme-based styles

carlobeltrame committed:
Fix e-input and api-input labels

BacLuc committed:
frontend: use jsonStringifyReactiveValue in generatePdfMixin.js

Was forgotten in commit 734bb4a.

BacLuc committed:
frontend: fix cannot call split: path is undefined in components/print/config

BacLuc committed:
frontend: fix format

BacLuc committed:
frontend: fix excel download in useMaterialViewHelper.js

BacLuc committed:
e2e: skip failing tests for now

That we can fix them one by one but have the safety net of the running tests.

BacLuc committed:
frontend: remove @vue/compat

Sadly, this does not lead to a change in the package-lock.json.

BacLuc committed:
frontend: skip failing tests

We don't have time right now to fix them, so we do it later.
At least we now if we broke one of the succeeding tests.

BacLuc committed:
frontend: skip even more failing tests

We don't have time right now to fix them, so we do it later.
At least we now if we broke one of the succeeding tests.

BacLuc committed:
Merge remote-tracking branch 'upstream/devel' into vue3

# Conflicts:
#	docker-compose.yml

pmattmann committed:
admin checklist, fix dragable

usu committed:
revert ETextField back to options API

BacLuc committed:
frontend: only send properties which might change via patch

To fix the update of ProgressLabels and MaterialLists.
With the update of vue, entityData is in some cases
a store value with the _meta key and the id field.
If you send all this with the patch request, the request fails.

usu committed:
fix ETextField ref

usu committed:
Merge remote-tracking branch 'ecamp/vue3' into vue3

BacLuc committed:
frontend: remove :item-key again from ApiSortable.vue

This reverts parts of commit f95ca35.
Rearranging contentnodes still worked and did not lead to console errors or warnings.

BacLuc committed:
frontend: fix validation for EAutocomplete.vue

BacLuc committed:
frontend: allow to open ChecklistCreate.vue

carlobeltrame committed:
Fix category editing

BacLuc committed:
frontend: set item-title and item-value as default in EAutocomplete.vue

The same as with ESelect.

pmattmann committed:
fix ChecklistItemCreate show Dialog

BacLuc committed:
frontend: fix autocomplete in EAutocomplete.vue

Of course, they changed the order of the parameters.
https://github.com/vuetifyjs/vuetify/blob/master/packages/vuetify/src/composables/filter.tsx#L22-L22

usu committed:
fix validation for ApiWrapper

simfeld committed:
fix control page table headers

simfeld committed:
remove console.log

BacLuc committed:
frontend: fix highlighting of autocomplete

BacLuc committed:
frontend: ignore whole key for no-unused-selector in SortableChecklistItem.vue

Fixes:
  102:1  warning  The selector `.e-sortable-checklist-item:is(:hover,:focus-visible):not(:has(.e-sortable-checklist-item:hover)):not(:has(.e-sortable-checklist-item__add:hover))>.v-list-item>.e-sortable-checklist-item__edit` is unused  vue-scoped-css/no-unused-selector

simfeld committed:
fix model-value prop naming

apiPropsMixin defines a modelValue prop which overrides model-value when destructuring with v-bind="{ ...$props, ...$attrs }"

simfeld committed:
Merge remote-tracking branch 'upstream/vue3' into vue3

usu committed:
Merge remote-tracking branch 'ecamp/vue3' into vue3

carlobeltrame committed:
Fix vertical spacing of inputs

carlobeltrame committed:
Fix delete dialog translation

BacLuc committed:
frontend: improve LAThematicArea

The select now opens but the value cannot be updated.

BacLuc committed:
frontend: fix eslint in ApiWrapper

validate is defined but never used.

BacLuc committed:
frontend: fix eslint in ETextField.vue

Property must be required or have a default value.

BacLuc committed:
frontend: fix deleting materiallist

Just use the materialList property which is known for sure.

BacLuc committed:
frontend: use model-value instead of v-model in ApiSortable.vue

Now a patch is sent after reordering the values.

BacLuc committed:
frontend: update localSortedItems after updating the endpoint in ApiSortable

It seems that the change is not detected in the handler -> we do it manually for now.

BacLuc committed:
frontend: update local selection and send update when toggling item in LAThematicArea.vue

BacLuc committed:
frontend: emit events to parent in ChecklistItem

Else only the top level items work.

BacLuc committed:
frontend: style ChecklistItem

Add padding to ol, this is not done by a global class anymore.
Overwrite grid-template-areas for v-switch that the switches are horizontally aligned.
Remove align-baseline that the list item and the v-switch are vertically aligned.

BacLuc committed:
frontend: fix ResetPassword

The minimal changes that it works again.
The password strength meter does not work yet.

BacLuc committed:
frontend: fix error of debouncedPasswordStrengthCheck in ResetPassword

Now the first parameter is the event, not the input.

BacLuc committed:
fixup! frontend: fix ResetPassword

BacLuc committed:
frontend: fix type check of dense failed in ResetPassword

BacLuc committed:
frontend: fix type coercion errors in ResetPassword

BacLuc committed:
frontend: fix error of debouncedPasswordStrengthCheck in Register

Now the first parameter is the event, not the input.

BacLuc committed:
frontend: show current email in Profile

BacLuc committed:
frontend: fix changeMailRequest in DialogChangeMail

BacLuc committed:
frontend: remove DialogActivityEdit

Did not work and is not necessary for the app to work.

BacLuc committed:
frontend: also set :model-value for Field

Else the validation doesn't seem to have the form value.
Seems to belong to commit 739752df.

BacLuc committed:
frontend: fix DialogPeriodForm

The @update:model-value seems strange, but it works.

BacLuc committed:
frontend: fix DialogPeriodDateEdit

Also fix all "==".

BacLuc committed:
frontend: fix ECheckbox.spec.js

Setting up vuetify is now simpler, it just needs the setupVuetify function at the start.
In setupVuetify we need the Vuetify plugin, not the vuetify instance.
It is now easier to make ad hoc components for tests, no Vue.component needed anymore.
You cannot use the wrapper for snapshots anymore, you need wrapper.html().

Vuetify internals: the checkbox always has the attribute value with the value "true".
The difference between checked and unchecked is the checked attribute.

I also did not find a way to fetch the current values of the data of a component.
That's why we have an additional span with the value of data.

I also could remove an unnecessary resetAllMocks.

BacLuc committed:
frontend: fix lint for ECheckbox.spec.js

carlobeltrame committed:
The path prop should not be required

carlobeltrame committed:
Update lockfile

simfeld committed:
Enable textarea and richtext on Controls page

carlobeltrame committed:
Fix PopoverPrompt display

carlobeltrame committed:
Set a default value which vuetify accepts

manuelmeister committed:
Fix stepper spacing

carlobeltrame committed:
Migrate route matcher calls

usu committed:
merge fix/arm64

manuelmeister committed:
Remove duplicate port assignment for frontends

carlobeltrame committed:
Fix styling of paste preview

BacLuc committed:
frontend: fix e-select and e-checkbox in print/config

They don't emit @input anymore, just @update:model-value.

BacLuc committed:
frontend: remove value again from v-stepper-item

This does not work manuel!!!
The steps now start with 0, see previous commits here.

manuelmeister committed:
Fix stepper icons

manuelmeister committed:
Fix stepper default opacity

BacLuc committed:
frontend: migrate v-list-item-icon and remove v-list-item-content

https://vuetifyjs.com/en/getting-started/upgrade-guide/#v-list

carlobeltrame committed:
Pass modelValue to vee-validate

manuelmeister committed:
Change secondary color

carlobeltrame committed:
Don't pass modelValue where it is not available

BacLuc committed:
frontend: migrate buttons in StoryboardRow(Default,Dense)

Use icon prop instead of child.
Use variant text for delete buttons.
The button position is still off.

usu committed:
use serializedValue for validation of EParseValue

simfeld committed:
Fix tiptap double update event and default props

BacLuc committed:
frontend: move contentComponents to computed in PrintConfigurator

Vue complained that components were in a reactive value...
they are still, but it doesn't complain anymore.

manuelmeister committed:
Add default options for buttons

BacLuc committed:
Merge remote-tracking branch 'upstream/devel' into vue3

manuelmeister committed:
Switch filled=false with variant=underlined

manuelmeister committed:
Adjust alert in camp create

manuelmeister committed:
Add gap to camp create card actions

manuelmeister committed:
Fix color of ContentGroup icon

manuelmeister committed:
Readd missing e-form wrapper

simfeld committed:
Avoid native input events

simfeld committed:
Fix textarea

BacLuc committed:
frontend: update ECheckbox.spec.js.snap

Changed in one of the commits before.

BacLuc committed:
frontend: fix lint

manuelmeister committed:
Style of material table

BacLuc committed:
frontend: CreateCampPeriods.vue comment out add button in CreateCampPeriods.vue

The validation does not yet work with multiple periods.

BacLuc committed:
api: ignore reference.php in .gitignore

This file changes too often.

manuelmeister committed:
Fix styling of add button

manuelmeister committed:
Fix spacing, styling of add button

carlobeltrame committed:
Declare actually emitted events

manuelmeister committed:
Fix dialog toolbar styling

manuelmeister committed:
Change default field color

simfeld committed:
Fix storyboard dragging by avoiding native events

manuelmeister committed:
Fix skeleton loader

simfeld committed:
Remove unused imports

carlobeltrame committed:
Fix intlify console warnings by providing a label

simfeld committed:
Fix button variant

carlobeltrame committed:
Tolerate more to avoid more intlify warnings

carlobeltrame committed:
No need to use toRefs on attrs

carlobeltrame committed:
Remove obsolete options

carlobeltrame committed:
Use scope global everywhere to avoid vue-i18n warning

BacLuc committed:
Merge remote-tracking branch 'upstream/devel' into vue3

# Conflicts:
#	docker-compose.yml

pmattmann committed:
fix DangerZone

manuelmeister committed:
Fix login button style

carlobeltrame committed:
Merge remote-tracking branch 'origin/devel' into vue3

carlobeltrame committed:
Improve alert spacing

simfeld committed:
Fix MaterialTable footer slot and header title

pmattmann committed:
Fix CollaboratorForm Role-Select

manuelmeister committed:
Update vuetify

manuelmeister committed:
Fix daily head day responsible

carlobeltrame committed:
Improve print configurator layout

usu committed:
fix BasePicker validation

carlobeltrame committed:
Migrate filled prop to variant

carlobeltrame committed:
Fix prop ordering

manuelmeister committed:
Add todo to rename fields to match EParseField

BacLuc committed:
e2e: activate zz-createCamp.cy.js

carlobeltrame committed:
Set underlined styles

manuelmeister committed:
Migrate to variant and density

manuelmeister committed:
Fix plain to outlined

manuelmeister committed:
Remove margin for the append fields

I'm not exactly sure if this applies everywhere, maybe we need to change this in the future

manuelmeister committed:
Fix changed v-col definition

carlobeltrame committed:
Pass on the variant to Vuetify

manuelmeister committed:
Remove duplicate variant definition

carlobeltrame committed:
Declare emitted events

carlobeltrame committed:
Add missing translation

carlobeltrame committed:
Fix lint

carlobeltrame committed:
Migrate dense prop to density

carlobeltrame committed:
Improve storyboard and rich text visuals

manuelmeister commented on frontend/src/components/checklist/ChecklistDetail.vue:
```diff
@@ -23,8 +23,7 @@
       <api-form v-if="editChecklistName" :entity="checklist" class="mx-2 flex-grow-1">
         <api-text-field
           path="name"
-          :disabled="layoutMode"
```
@carlobeltrame why remove disabled?

carlobeltrame replied:
Because there is no layoutMode on this view. This line caused a console error.

manuelmeister commented on frontend/src/components/form/tiptap/TiptapEditor.vue:
```diff
@@ -396,6 +396,9 @@ div.editor:deep(.editor__content .ProseMirror li p) {
 div.editor:deep(.editor__content .ProseMirror li p:not(:last-child)) {
   margin-bottom: 0;
 }
+.editor.editor--editable {
+  cursor: text;
+}
```
@carlobeltrame why is this necessary?

carlobeltrame replied:
The cursor wasn't right otherwise. Don't know why.

simfeld committed:
Fix MaterialTable edit button style

carlobeltrame committed:
Fix print config delete button styling

carlobeltrame committed:
Disable print preview frame on prod

carlobeltrame committed:
Fix print filter display

carlobeltrame committed:
Declare emits

carlobeltrame committed:
Cache more of the repeated filter calculation

BacLuc committed:
Profile: revert commit 5ce79ab7

This was clearly not tested.

BacLuc committed:
Register.vue: revert commit 5ce79ab7

This was clearly not tested.

BacLuc committed:
Controls.vue: revert commit 5ce79ab7

This was clearly not tested.

BacLuc committed:
frontend: listen to @update:model-value instead of @input in DayResponsibles.vue

Else we don't detect changes.

BacLuc committed:
frontend: also emit input event in ESelect.vue

We are not very good at doing changes over the whole codebase.
e.g. activityResponsibles did still listen to @input.
-> lets make a soft migration to @update:model-value.

BacLuc committed:
frontend: debounce the edit mode a little in SortableChecklist

Else the browser crashes and cannot render this page.

BacLuc committed:
Merge remote-tracking branch 'upstream/devel' into vue3

BacLuc committed:
frontend: delay showing edit controls until loaded in CategoryTemplate.vue

Else the tab breaks, but only in production deployment.
Build preview does not reproduce the error.
Same trick as in commit e24e8cc.

BacLuc committed:
Revert "frontend: delay showing edit controls until loaded in CategoryTemplate.vue"

This reverts commit e7384536dc4ec24737f28e77501f07d250b3d027.

BacLuc committed:
frontend: cache rootContentNode and children to fix performance issue in category template edit view

Error:
Uncaught (in promise) Maximum recursive updates exceeded in component <CategoryTemplate>. This means you have a reactive effect that is mutating its own dependencies and thus recursively triggering itself. Possible sources include component template, render function, updated hook or watcher source function.

This happened only in production.

AI said:
Reviewed the components involved: CategoryTemplate.vue, RootNode.vue, ContentNode.vue, ColumnLayout.vue, and ResponsiveLayout.vue. Noted CategoryTemplate repeatedly calls category.rootContentNode() directly in the template in multiple places, which can cause side effects during render and trigger recursive updates, especially in production. Also identified ResponsiveLayout’s injected isPaperDisplaySize() usage which could be impure if provider mutates state. Hypothesis: recursive loop originates from CategoryTemplate’s render invoking a side-effectful getter repeatedly, cascading through RootNode -> activity/content components.

Why dev works but prod freezes
•
In dev, repeated reads may be coalesced and not trip the maximum update threshold as easily.
•
In prod, the optimizer can make the repeated method calls happen tighter in the render loop, so a side-effect inside rootContentNode()/children() can immediately re-trigger the same render.

BacLuc committed:
fixup! frontend: cache rootContentNode and children to fix performance issue in category template edit view

The error still persists.

BacLuc committed:
Revert "frontend: cache rootContentNode and children to fix performance issue in category template edit view"

This reverts commit 4c8598ccad7a669a2b0b1b2adc0a8cacac86f816 and 9c182b0a1ba4d011003c9b2d32f447d1922b63c2

BacLuc committed:
frontend: switch to conditional rendering instead of v-show in CategoryTemplate.vue

The browser crashes if we load both at the same time with vue3.

manuelmeister committed:
Add missing padding for list

manuelmeister committed:
Fix icon size of storyboard row buttons

manuelmeister committed:
Fix storyboard row baseline align

BacLuc committed:
frontend: don't make editor deeply reactive in TipTapEditor.vue

This fixes the crashes when typing.

BacLuc committed:
frontend: declare 'activity-created' event in DialogActivityCreate

BacLuc committed:
frontend: revert "fixed eslint @vue/no-mutating-props"

This doesn't work anymore, the category is detached after the form is submitted the first time.
I will create an issue to redo this with emits.

BacLuc committed:
frontend: declare 'activity-updated' event in DialogActivityEdit

BacLuc committed:
Revert "frontend: don't make editor deeply reactive in TipTapEditor.vue"

This reverts commit ea068260b891bd8a84fbaa1973805ca9e2c802ad.

BacLuc committed:
frontend: fix warning [Vue warn]: Property "value" was accessed during render but is not defined on instance. in DialogActivityCreate

A leftover :value.

BacLuc committed:
frontend: workAround for silly call to modelValue watch in TiptapEditor.vue

BacLuc committed:
frontend: manually reset MateriaTable form after submit

BacLuc committed:
frontend: disable Pagination in MaterialTable.vue

They changed the api.

BacLuc committed:
frontend: fix warning that role is already defined in attrs appearing in PrintConfigurator.vue

BacLuc committed:
frontend: mark campRole as private in campRoleMixin.js

It is currently only used here.

BacLuc committed:
frontend: fix drag and drop for PrintConfigurator.vue

We need to wrap the draggable slot (copied from ApiSortable).
PagesOverview now emits @update:model-value and not @input.

BacLuc committed:
frontend: simplify item-key in PagesOverview.vue

It anyway always resulted in undefined, there is no index passed to this function.

BacLuc committed:
Revert "frontend: debounce the edit mode a little in SortableChecklist"

This reverts commit e24e8cc11ff8f8a0929dcae19375fda2b68c318f.

BacLuc committed:
frontend: debounce the edit mode a little in ChecklistDetail.vue

Else the browser crashes and cannot render this page.

BacLuc committed:
frontend: remove the "required" validation from FormScheduleEntryItem

This prohibits the form from beeing submitted once a date or a time is changed via picker.

carlobeltrame committed:
Merge remote-tracking branch 'origin/devel' into vue3

BacLuc committed:
frontend: show filter chip in toolbar items when not activated in CampProgram

Now v-chip is not shown when model-value is false.

BacLuc committed:
frontend: allow to scroll on mobile in SortableChecklistItem

Editing the item is still not possible, but the same issue was in
the old frontend.

manuelmeister committed:
Fix ScheduleEntryList

manuelmeister committed:
Fix design of ColorSwatch

manuelmeister committed:
Fix design of ColorPicker popup

manuelmeister committed:
Fix ColorPicker functionality

manuelmeister committed:
Fix profile

manuelmeister committed:
Fix icon inner position in change mail dialog

BacLuc committed:
Revert "frontend: allow to scroll on mobile in SortableChecklistItem"

This reverts commit f9aed2b2c5d664ec089478d3c1b9085f5bec4ee7.

BacLuc committed:
frontend: fix lint for commit 6b04e99

BacLuc committed:
renovate.json: disable renovate for frontend-old

manuelmeister commented:
## Core Meeting Decision

From now on we will work again with PRs.
